test/jest/Bladeburner/Actions.test.ts

```diff
@@ -0,0 +1,358 @@
+      let beforeMinor, minorGain, beforeMajor, majorGain;
+
+      it.each([
+        { major: SampleBlackOp, minor: SampleOperation },
+        { major: SampleOperation, minor: SampleContract },
+      ])("$major.type reward significantly more rank than $minor.type", ({ major, minor }) => {
+        guaranteeSuccess(), (beforeMinor = inst.rank), complete(minor), finish(), (minorGain = inst.rank - beforeMinor);
+        guaranteeSuccess(), (beforeMajor = inst.rank), complete(major), finish(), (majorGain = inst.rank - beforeMajor);
+        expect(majorGain).toBeGreaterThan(minorGain);
+      });
+    });
+
+    describe(BladeburnerGeneralActionName.InciteViolence, () => {
+      const Incite = GeneralAction.createId(BladeburnerGeneralActionName.InciteViolence);
+      let chaos;
+
+      it("generates available contracts", () => {
+        const { count } = inst.getActionObject(SampleContract);
+        guaranteeSuccess(), complete(Incite);
+        expect(inst.getActionObject(SampleContract).count).toBeGreaterThan(count);
+      });
+
+      it("generates available operations", () => {
+        const { count } = inst.getActionObject(SampleOperation);
+        guaranteeSuccess(), complete(Incite);
+        expect(inst.getActionObject(SampleOperation).count).toBeGreaterThan(count);
+      });
+
+      /** Relates to all issues mentioned in PR-1586 */
+      it.each(CITIES)("SIGNIFICANTLY increases chaos in all cities when chaos is LOW: %s", (city: CityName) => {
+        guaranteeSuccess(), ({ chaos } = inst.cities[city]), complete(Incite);
+        expect(inst.cities[city].chaos).toBeGreaterThan(chaos * 2);
+      });
+
+      /** Relates to all issues mentioned in PR-1586 */
+      it.each(CITIES)("MILDLY increases chaos in all cities when chaos is HIGH: %s", (city: CityName) => {
+        guaranteeSuccess(), allCitiesHighChaos(), ({ chaos } = inst.cities[city]), complete(Incite);
+        expect(inst.cities[city].chaos).toBeGreaterThan(chaos * 1.05);
+      });
+    });
+
+    describe(BladeburnerGeneralActionName.Recruitment, () => {
+      const Recruitment = GeneralAction.createId(BladeburnerGeneralActionName.Recruitment);
+
+      it("awards charisma exp", () => {
+        guaranteeSuccess(), (before = Player.exp.charisma), complete(Recruitment);
+        expect(Player.exp.charisma).toBeGreaterThan(before);
+      });
+
+      it("hires team member", () => {
+        guaranteeSuccess(), complete(Recruitment);
+        expect(inst.teamSize).toBeGreaterThan(0);
+      });
+    });
+
+    describe.each([...actionId(contracts())])("$id.name", ({ id }) => {
+      it("all contracts award money", () => {
+        guaranteeSuccess(), (before = Player.money), complete(id);
+        expect(Player.money).toBeGreaterThan(before);
+      });
+    });
+
+    /** Stat EXP check for all actions */
+    /** Checking all of them to avoid regressions */
+    describe.each([...actionIdWithIndividualStat(nonGeneralActions())])("$id.name", ({ id, stat }) => {
```

Comment:
This is the most complex bit. It generates pairs of Action + Stat depending on which stat weights are non-zero for each given action. This allows for a quick rundown of all possible combos to make sure we're not missing awards for things like hacking exp and int exp.

These are not hardcoded to the Actions per se, but using the data files through the static Bladeburner instance. It is somewhat coupled to implementation detail but I figured it was the lesser of two evils rather than writing a test suite of all the combinations manually.